urnx/validate.go

```diff
@@ -0,0 +1,24 @@
+package urnx
+
+import (
+	"regexp"
+)
+
+func validateNamespace(namespace string) (bool, error) {
+	rx, err := regexp.MatchString("^[A-Za-z0-9-]{1,30}$", namespace)
+	if err != nil {
+		return false, err
+	}
+
+	return rx, nil
+}
+
+func validateResourceType(name string) (bool, error) {
+	rx, err := regexp.MatchString("^[A-Za-z0-9-]{1,}$", name)
```

Comment:
Setting _some_ limit on resource type length is probably sound, since right now I could potentially send a huge payload here and get it to validate.